scripts/jenkins/jenkins-job-trigger

```diff
@@ -1,96 +1,71 @@
-  print STDERR "Please create your own crumb:\n";
-  print STDERR " curl --user <myusername>:<mypassword> 'https://ci.opensuse.org/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,\":\",//crumb)'";
-  print STDERR "And then add it to your jenkinsapi.cred file:\n";
-  print STDERR ' $CRUMB=".crumb:123abc456def"';
-}
-
-$METHOD ||= 'POST';
-$SCHEME ||= 'https';
-if ($JURL =~ /river\.suse\.de/)
-{
-  $SCHEME='http';
-}
-
-my $up='';
-$up="${USER}:${APIKEY}\@" if (defined $USER && $USER ne '' && defined $APIKEY && $APIKEY ne '');
-my $jurl="$SCHEME://${up}${JURL}";
-my $job=shift || die usage();
-
-my $jobtype='normal';
-my $jtype=shift || '';
-if ($jtype =~ /^-p$/) {
-  $jobtype='parameter';
-} elsif ($jtype =~ /^-m$/) {
-  $jobtype='matrix';
-}
-
-my $triggerurl=$jurl."/job/".$job;
-my @curlargs=('--silent', '--show-error');
-push @curlargs, ("-H", "Content-Length: 0");
-
-if ($jobtype eq 'normal') {
-  $triggerurl .= "/build";
-} elsif ($jobtype eq 'parameter') {
-  $triggerurl .= "/buildWithParameters?".join('&', @ARGV);
-} elsif ($jobtype eq 'matrix') {
-  $triggerurl .= "/".join(',', @ARGV)."/build";
-}
-
-$triggerurl =~ s/ /+/g;
-push @curlargs, ("-X", $METHOD);
-push @curlargs, ("-H", $CRUMB) if ($CRUMB);
-my @cmd=("curl", @curlargs, $triggerurl);
-
-print join(' ', @cmd) if ( $ENV{debug} );
-exec(@cmd) || die $!;
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+# Triggers a jenkins job via the API
+# Needed for job chains that are not possible to implement natively
+
+import jenkins
+import json
+import os
+import sys
+
+
+def usage():
+    print("Error: No job name defined as first parameter.\n" +
+              "Add parameters as needed:\n" +
+              "  jenkins-job-trigger <job_name> [ para1=val1 [ para2=val2 ]... ]")
+    sys.exit(1)
+
+def jenkins_build_job(job_name, job_args=[]):
+    if not job_name:
+        usage()
+
+    config_files = ('/etc/jenkinsapi.conf', './jenkinsapi.conf')
+    config = dict()
+    job_parameters = dict()
+
+    for config_file in config_files:
+        if not os.path.exists(config_file):
+            continue
+        with open(config_file, 'r') as f:
+            config.update(json.load(f))
+
+    if not config:
+        print('Error: No config file could be loaded. Please create either of: %s' %
+              ', '.join(config_files))
+        sys.exit(1)
+
+    for param in job_args:
+        # backwards compatibility, ignore -p and -m
+        if param in ('-p', '-m'):
+            continue
+        p_key, p_val = param.split('=', 1)
+        job_parameters[p_key] = p_val
```

Comment:
are you sure that there is not an space in `p_key` or in `p_val`? Also the `partition` works as expected (as noted in another comment)